tools/ci_build/github/azure-pipelines/templates/react-native-ci.yml

```diff
@@ -266,42 +294,47 @@ stages:
-            options: '--stacktrace'
-            tasks: ':app:connectedDebugAndroidTest'
-            publishJUnitResults: true
-            testResultsFiles: '**/TEST-*.xml'
-            testRunTitle: 'React Native Android e2e Test results'
-            javaHomeOption: 'path'
-            jdkDirectory: '$(JAVA_HOME_11_X64)'
-            sonarQubeRunAnalysis: false
-            spotBugsAnalysis: false
-          displayName: Run React Native Android e2e Tests
-
     - script: |
-        export FORCE_BUNDLING=1
-        export RCT_NO_LAUNCH_PACKAGER=1
-        export ENTRY_FILE=index.tsx
-        xcrun xcodebuild test -workspace $(Build.SourcesDirectory)/js/react_native/e2e/ios/OnnxruntimeModuleExample.xcworkspace \
-          -scheme OnnxruntimeModuleExample -destination 'platform=iOS Simulator,OS=latest,name=iPhoneRNTest' \
-          -derivedDataPath $(Build.BinariesDirectory)/react_native/ios_e2e_test/derived_data | xcpretty -r junit --no-color \
-          --output $(Build.SourcesDirectory)/js/react_native/e2e/ios/build/reports/test-results.xml
+        detox build --configuration android.emu.release
+      workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+      displayName: Build React Native Detox Android e2e Tests
+  
+    - script: |
+        detox test --record-logs all  --configuration android.emu.release
       workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
-      displayName: Run React Native iOS e2e tests
+      displayName: Run React Native Detox Android e2e Tests
+    
+    - task: PublishTestResults@2
+      inputs:
+        testResultsFiles: '$(Build.SourcesDirectory)/js/react_native/e2e/junit.xml'
+        failTaskOnFailedTests: true
+        testRunTitle: 'React Native Detox Android e2e Test Results'
+      condition: succeededOrFailed()
+      displayName: Publish React Native Detox Android e2e Test Results
+    
+    - script: |
+        detox build --configuration ios.sim.release
+      workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+      displayName: Build React Native Detox iOS e2e Tests
+  
+    - script: |
+        detox test --record-logs all --configuration ios.sim.release
+      workingDirectory: '$(Build.SourcesDirectory)/js/react_native/e2e'
+      displayName: Run React Native Detox iOS e2e Tests
 
     - task: PublishTestResults@2
       inputs:
-        testResultsFiles: '$(Build.SourcesDirectory)/js/react_native/e2e/ios/build/reports/test-results.xml'
+        testResultsFiles: '$(Build.SourcesDirectory)/js/react_native/e2e/junit.xml'
         failTaskOnFailedTests: true
-        testRunTitle: 'React Native iOS e2e Test results'
+        testRunTitle: 'React Native Detox iOS e2e Test Results'
       condition: succeededOrFailed()
-      displayName: Publish React Native iOS e2e Test Results
+      displayName: Publish React Native Detox iOS e2e Test Results
+    
+    - task: DownloadPipelineArtifact@2
```

Comment:
this is downloading onto the build agent in $(Build.SourcesDirectory)/js/react_native/e2e/artifacts. did you want to upload the contents of $(Build.SourcesDirectory)/js/react_native/e2e/artifacts instead?
see this task: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-pipeline-artifact-v1?view=azure-pipelines

Reply:
ah ok, sorry not too familiar with the pipeline tasks... thanks!

Reply:
I think the DownloadPipelineArtifact step is not needed here

Reply:
yes, removed.